app/components/form/image-upload-avatar.tsx

```diff
@@ -0,0 +1,70 @@
+import { FieldConfig } from "@conform-to/react";
+import { PlusIcon } from "lucide-react";
+import React, { ChangeEvent, useState } from "react";
+import { z } from "zod";
+import { FileSchema } from "~/services/schemas.server";
+import { cn } from "~/util/utils";
+
+export const ImageUploadAvatar = React.forwardRef<
+  HTMLImageElement,
+  React.ImgHTMLAttributes<HTMLImageElement> & {
+    fieldProps: FieldConfig<z.infer<typeof FileSchema>>;
+  }
+>(({ className, fieldProps, src, ...props }, ref) => {
+  const [previewImage, setPreviewImage] = useState<string | null>(src || null);
+  return (
+    <>
+      <label>
+        {previewImage ? (
+          <img
+            className={cn("cursor-pointer hover:opacity-80", className)}
+            {...props}
+            src={previewImage}
+            ref={ref}
+          />
+        ) : (
+          <div
+            className={cn(
+              "flex cursor-pointer items-center justify-center bg-accent hover:opacity-80",
+              className
+            )}
+          >
+            <PlusIcon size={24} />
+          </div>
+        )}
+        <input
+          type="file"
+          className="sr-only"
+          onChange={async (event: ChangeEvent<HTMLInputElement>) => {
+            const file = event.target.files?.[0];
+            if (file) {
+              const data = new FormData();
+              data.append("file", file);
+              data.append("upload_preset", "bidhub_user_avatar");
+
+              const res = await fetch(
```

Comment:
I couldn't get the cloudinary package to work on the client so going to stick with this until I come up with a server solution.